### PR TITLE
Component datums

### PR DESCRIPTION
**Pull request by DamianX**
- Commented out the old component system, it will have to be ported to the new one at a later time
- This is still not used by anything

### CODE REVIEW

File: code/datums/components/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## API
+
+### Defines
+
+1. `COMPONENT_INCOMPATIBLE` Return this from `/datum/component/Initialize` or `datum/component/OnTransfer` to have the component be deleted if it's applied to an incorrect type. `parent` must not be modified if this is to be returned. This will be noted in the runtime logs
+
+### Vars
+
+1. `/datum/var/list/datum_components` (private)
+    * Lazy associated list of type -> component/list of components.
```
Comment (ShiftyRail) [inline code/datums/components/README.md]: >Lazy

Comment (ShiftyRail) [inline code/datums/components/README.md:17]: >Lazy

Comment (DamianX) [reply code/datums/components/README.md:17]: it is lazy, yes

**Comment by gbasood** [comment]
>Commented out the old component system, it will have to be ported to the new one at a later time

Don't simple animals use some kind of component system?

**Comment by DamianX** [comment]
We have some mobs using component AI but they aren't spawned/used by the game, as far as I know.
@MadmanMartian is that correct?

**Comment by MadmanMartian** [comment]
It is correct, there are two mobs using component systems right now. Giant spiders and NPC humans. Neither are currently spawned in on maps.

**Comment by gbasood** [comment]
What? Russian mobs don't show up anywhere anymore, even in vaults?
What about giant spiders, which can show up in round at a button press?

**Comment by DamianX** [comment]
Russian mobs have nothing do with component AI, same with giant spiders, they are a separate type.
Nonetheless, there's no feature loss on the server because **this is not Bleeding-Edge**

**Pull request merged**